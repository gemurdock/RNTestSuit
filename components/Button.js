import { Pressable, StyleSheet, Text } from "react-native"
import { APP_COLORS } from "../lib/colors"

const Button = ({ children, onPress }) => {
    return (
        <Pressable onPress={onPress} style={styles.button}>
            <Text style={styles.text}>
                {children}
            </Text>
        </Pressable>
    )
}

const styles = StyleSheet.create({
    button: {
        backgroundColor: APP_COLORS.primary,
        padding: 10,
        borderRadius: 10,
    },
    text: {
        color: 'white'
    }
})

export default Button