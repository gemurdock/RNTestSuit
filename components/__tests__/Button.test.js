import { render } from '@testing-library/react-native'
import Button from '../Button'

test('Button renders correctly', () => {
    const tree = render(<Button>Test</Button>).toJSON()
    expect(tree).toMatchSnapshot()
})
