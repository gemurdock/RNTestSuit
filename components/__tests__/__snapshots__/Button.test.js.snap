// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#296894",
      "borderRadius": 10,
      "padding": 10,
    }
  }
>
  <Text
    style={
      {
        "color": "white",
      }
    }
  >
    Test
  </Text>
</View>
`;
